File: mobile/lib/shared/models/person.dart
// 人物实体，对齐 frontend/packages/api/src/modules/history-types.ts 的 Person。

import 'package:flutter/foundation.dart';

@immutable
class Person {
  final int id;
  final String name;
  final String courtesyName;
  final String aliasName;
  final int dynastyId;
  final int birthYear;
  final int deathYear;
  final String gender;
  final String title;
  final String biography;
  final String achievements;
  final String portraitUrl;

  const Person({
    required this.id,
    required this.name,
    this.courtesyName = '',
    this.aliasName = '',
    this.dynastyId = 0,
    this.birthYear = 0,
    this.deathYear = 0,
    this.gender = '',
    this.title = '',
    this.biography = '',
    this.achievements = '',
    this.portraitUrl = '',
  });

  factory Person.fromJson(Map<String, dynamic> json) {
    return Person(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      courtesyName: json['courtesy_name'] as String? ?? '',
      aliasName: json['alias_name'] as String? ?? '',
      dynastyId: json['dynasty_id'] as int? ?? 0,
      birthYear: json['birth_year'] as int? ?? 0,
      deathYear: json['death_year'] as int? ?? 0,
      gender: json['gender'] as String? ?? '',
      title: json['title'] as String? ?? '',
      biography: json['biography'] as String? ?? '',
      achievements: json['achievements'] as String? ?? '',
      portraitUrl: json['portrait_url'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'courtesy_name': courtesyName,
        'alias_name': aliasName,
        'dynasty_id': dynastyId,
        'birth_year': birthYear,
        'death_year': deathYear,
        'gender': gender,
        'title': title,
        'biography': biography,
        'achievements': achievements,
        'portrait_url': portraitUrl,
      };
}

File: backend/history-service/internal/application/dto/person.go
package dto

// PersonRequest is the create/update payload for history_person.
type PersonRequest struct {
	Name         string `json:"name"`
	CourtesyName string `json:"courtesy_name,omitempty"`
	AliasName    string `json:"alias_name,omitempty"`
	DynastyID    int64  `json:"dynasty_id,omitempty"`
	BirthYear    int16  `json:"birth_year,omitempty"`
	DeathYear    int16  `json:"death_year,omitempty"`
	Gender       string `json:"gender,omitempty"`
	Title        string `json:"title,omitempty"`
	Biography    string `json:"biography,omitempty"`
	Achievements string `json:"achievements,omitempty"`
	PortraitURL  string `json:"portrait_url,omitempty"`
}

// PersonResponse is the wire representation of a person.
type PersonResponse struct {
	ID           int64  `json:"id"`
	Name         string `json:"name"`
	CourtesyName string `json:"courtesy_name"`
	AliasName    string `json:"alias_name"`
	DynastyID    int64  `json:"dynasty_id"`
	BirthYear    int16  `json:"birth_year"`
	DeathYear    int16  `json:"death_year"`
	Gender       string `json:"gender"`
	Title        string `json:"title"`
	Biography    string `json:"biography"`
	Achievements string `json:"achievements"`
	PortraitURL  string `json:"portrait_url"`
}
